lib/rucio/common/utils.py

```diff
@@ -618,9 +625,42 @@ def register_surl_algorithm(surl_callable, name=None):
 register_surl_algorithm(construct_surl_BelleII, 'BelleII')
 
 
+def _register_policy_package_surl_algorithms():
+    from rucio.common import config
+    from rucio.core.vo import list_vos
+    import importlib
+    try:
+        multivo = config.config_get_bool('common', 'multi_vo')
+    except (NoOptionError, NoSectionError):
+        multivo = False
+    if not multivo:
+        # single policy package
+        try:
+            package = config.config_get('policy', 'package')
+            module = importlib.import_module(package)
+            if hasattr(module, 'get_surl_algorithms'):
+                _SURL_ALGORITHMS.update(module.get_surl_algorithms())
+        except (NoOptionError, NoSectionError, ImportError):
+            pass
+    else:
+        # policy package per VO
+        vos = list_vos()
+        for vo in vos:
+            try:
+                package = config.config_get('policy', 'package-' + vo['vo'])
+                module = importlib.import_module(package)
+                if hasattr(module, 'get_surl_algorithms'):
+                    _SURL_ALGORITHMS.update(module.get_surl_algorithms())
+            except (NoOptionError, NoSectionError, ImportError):
+                pass
```

Comment:
I have factored out the common code into a function as suggested. Yes, it is true that different VO policies can override each other. The documentation instructs policy package authors to ensure their function names are prefixed with the VO name to avoid conflicts. Maybe this should also be enforced by the code? (it wasn't possible with the old system, but it would be with this one).

Reply:
> The documentation instructs policy package authors to ensure their function names are prefixed with the VO name to avoid conflicts.

that makes sense and it sounds good enough for this PR :+1:

It sounds like a good idea to enforce it (in another PR), thoughts @bari12 ?